tests/integration/test_api_keys_routes.py

```
@@ -38,6 +38,244 @@ def test_save_api_keys_and_read_back(client, monkeypatch, tmp_path):
     assert cfg.load_env_key("NASA_SECRET") == "route-test-123"
 
 
+def test_save_api_keys_empty_value_preserves_existing(client, monkeypatch, tmp_path):
+    """JTN-598: an empty posted value must not overwrite the existing .env entry."""
+    monkeypatch.setenv("PROJECT_DIR", str(tmp_path))
+    src_dir = Path(__file__).resolve().parents[2] / "src"
+    spec = importlib.util.spec_from_file_location("config", str(src_dir / "config.py"))
+    assert spec and spec.loader
```

Ruff (0.15.9)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)